apps/codeforafrica/src/lib/api.netlify-cms/utils.js

```diff
@@ -0,0 +1,29 @@
+import fs from "fs";
+import { join } from "path";
+
+import matter from "gray-matter";
+import { marked } from "marked";
+
+export const getCollectionSlugs = (collectionDir) => {
+  return fs.readdirSync(collectionDir);
+};
+
+export const getCollectionBySlug = (collectionDir, slug, fields = []) => {
+  const realSlug = slug.replace(/\.md$/, "");
+  const fullPath = join(collectionDir, `${realSlug}.md`);
+  const fileContents = fs.readFileSync(fullPath, "utf8");
+  const { data, content } = matter(fileContents);
+  const items = {};
+
+  fields.forEach((field) => {
+    if (field === "content") {
+      items.description = marked(content);
+    } else if (field === "slug") {
+      items.slug = realSlug;
+    } else {
+      items[field] = data[field] || null;
+    }
+  });
```

Comment:
1. We still assign `items.description = marked(content);`, are we sure **_every_** collection is like that? For example **imprint** page, `content` is `content`.
2. `reduce` is perfect here:
```suggestion
  const items = fields.reduce((acc, curr) => {
    if (curr === "content") {
      acc["content"] = marked(content);
    } else if (curr === "slug") {
      acc["slug"] = realSlug;
    } else {
      acc[field] = data[field] || null;
    }
    return acc;
  }, {});
```